src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -633,94 +633,64 @@ class npc_death_knight_initiate : public CreatureScript
 enum DarkRiderOfAcherus
 {
     SAY_DARK_RIDER              = 0,
-    SPELL_DESPAWN_HORSE         = 51918
-};
 
-class npc_dark_rider_of_acherus : public CreatureScript
-{
-    public:
-        npc_dark_rider_of_acherus() : CreatureScript("npc_dark_rider_of_acherus") { }
+    EVENT_START_MOVING          = 1,
+    EVENT_DESPAWN_HORSE         = 2,
+    EVENT_END_SCRIPT            = 3,
 
-        struct npc_dark_rider_of_acherusAI : public ScriptedAI
-        {
-            npc_dark_rider_of_acherusAI(Creature* creature) : ScriptedAI(creature)
-            {
-                Initialize();
-            }
-
-            void Initialize()
-            {
-                PhaseTimer = 4000;
-                Phase = 0;
-                Intro = false;
-                TargetGUID.Clear();
-            }
+    SPELL_DESPAWN_HORSE         = 52267
+};
 
-            void Reset() override
-            {
-                Initialize();
-            }
+struct npc_dark_rider_of_acherus : public ScriptedAI
+{
+    npc_dark_rider_of_acherus(Creature* creature) : ScriptedAI(creature)
+    {
+        if (TempSummon* summon = me->ToTempSummon())
+            _horseGUID = summon->GetSummonerGUID();
 
-            void UpdateAI(uint32 diff) override
-            {
-                if (!Intro || !TargetGUID)
-                    return;
+        _events.ScheduleEvent(EVENT_START_MOVING, 1s);
```

Comment:
it need be done inside of hooks like: Reset() or JustAppeared(), etc.
Not in constructor.